<a href="https://colab.research.google.com/github/mpfoster/Biochem5721/blob/master/Example_2_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Biochem 5721
## Example solving a problem a Jupyter Notebook
### Example 2.3 from Tinoco _et al._ 
Below are three different ways you might implement a solution in your Jupyter notebook.

### Problem:
Calculate the heat (in J) necessary to increase the temperature of 100.0 g of liquid water from 25.0°C to 75.0°C at ambient pressure (1 bar). The molar heat capacity of liquid water is 75.4 J mol<sup>-1</sup> K<sup>-1</sup> and is nearly independent of temperature.

_(This is the end of the markdown cell. In the following code cell, we show our approach and solution.)_

### Solution:

In [ ]:
# Use either "#" at the beginning of each line...
'''
... or the triple single-quotes to enter comments -- i.e., writing what is not intended
to be interpreted as code
'''
# Solution below
#---------------
# 100 g of water
# heat capacity of water: 75.4 J/mol/K
# temperature change: 25 to 75 C = 50 K
# heat = (mol water) x (heat capacity) x (temperature change)
# MW of water H2O is ~ 16+2 = 18 g/mol
mol_water = 100 / 18. # g ÷ g/mol
print("100 g of water is %5.2f mol" % mol_water)
heat = mol_water * 75.4 * 50. # J
print("Heating 100 g of water by 50 degrees takes %.3g J" % heat)

100 g of water is  5.56 mol
Heating 100 g of water by 50 degrees takes 2.09e+04 J


------

In [ ]:
# Another possible way to code the solution:
#---------------
mass_water = 100 # g
Cp_water =  75.4 # J/mol/K
delta_T = 75 - 25 # C = K
MW_water = 18  # g/mol
heat = mass_water/MW_water * Cp_water * delta_T  # g/g/mol x J/mol•K x K
print("The heat required to raise the temperature of {0:.1f} g \
of water by {1:.1f} K is {2:.0f} J".format(mass_water,delta_T,heat))

The heat required to raise the temperature of 100.0 g of water by 50.0 K is 20944 J


---
But, what if you don't want to mess with formatted printing, or use multiple code cells, instead of one? That's fine too. Your solution could look like this:

In [3]:
# MW water is 18 g/mol
# so, 100 g of water is:
100/18

5.555555555555555

In [2]:
# heat = mol x molar heat capacity (J/mol/K) x temperature change (K)
# temperature change is 75-25 = 50 K
5.55 * 75.4 * 50

20923.5

So, the answer is 20,923 J, or 20.9 kJ

---
One more simple example, using variable declarations:

In [5]:
# MW water is 18 g/mol
# so, 100 g of water is:
mol = 100/18
mol

5.555555555555555

In [6]:
# heat = mol x molar heat capacity (J/mol/K) x temperature change (K)
# temperature change is 75-25 = 50 K
dT = 75-25
heat = mol * 75.4 * dT
heat

20944.444444444445

So, the answer is 20.9 kJ